### PR TITLE
Modify forecasts to use site_id

### PR DESCRIPTION
**Pull request by Mnoble-19**
**_WHAT DOES THIS PR DO?_**
* Updates the forecast job to use the events api, instead of accessing the events Database
* Updates the hourly_forecast and daily_forecast endpoints, to use site-id and query parameters

**_HOW DO I TEST OUT THIS PR?_**
* Follow the steps outlined in the api/ directory's README.

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
* {baseUrl}/api/v2/daily-forecast?site_id={site_id}

First screenshot shows when the deploy preview is used. There are currently no forecasts because the updated forecast job has not been merged.
Screenshot 2023-04-09 at 12 01 29

This is what the endpoint should however return
Screenshot 2023-04-09 at 12 16 18

### CODE REVIEW

File: src/predict/jobs/forecast/models.py [hunk 1]
```diff
@@ -1,47 +1,58 @@
...
-            events_data = {}
-
-        return events_data
+            response = requests.get(Events.events_url, params=params)
+            if response.status_code != 200:
+                raise Exception(f'Error fetching data from events api: {response.text}')
+            else:
+                return response.json()
+
+        with concurrent.futures.ThreadPoolExecutor(max_workers=4) as executor:
```
Comment (Mnoble-19) [inline src/predict/jobs/forecast/models.py:41]: @Baalmart as per your suggestion. Data is now fetched from events api, in batches

File: src/predict/api/controllers/prediction.py [hunk 2]
```diff
@@ -64,30 +46,13 @@ def get_next_24hr_forecasts(device_channel_id, forecast_start_time):
...
-                return jsonify(error, 400)
-
-        # change prediction_start_date to datetime format
-        forecast_start_timestamp = dt.datetime.fromtimestamp(
-            forecast_start_date).isoformat()
-
-        print(forecast_start_timestamp)
-        result = get_next_1_week_forecasts_for_channel(
-            device_channel_id, forecast_start_timestamp)
+        site_id = request.args.get('site_id', type=str)
```
Comment (Mnoble-19) [inline src/predict/api/controllers/prediction.py:54]: @NoahNsimbe as per your suggestion, the endpoint has been updated to use query parameters

Comment (NoahNsimbe) [reply src/predict/api/controllers/prediction.py:54]: thanks @Mnoble-19

**Review by NoahNsimbe — APPROVED** [review]
LGTM. Thanks @Mnoble-19

File: src/predict/jobs/forecast/models.py [hunk 3]
```diff
@@ -1,47 +1,58 @@
...
-            'values': {
-                'pm2_5': 1,
-                'device_number': 1,
-                'time': 1
+    events_url = f"{configuration.AIRQO_API_BASE_URL}devices/events"
+
+    @staticmethod
+    def fetch_data_from_events_api():
+        """gets data from the events api in batches"""
+        start_date = datetime.now() - timedelta(days=30)
```
Comment (sserurich) [inline src/predict/jobs/forecast/models.py:17]: can we read this days=30 from env variable

File: src/predict/jobs/forecast/models.py [hunk 4]
```diff
@@ -1,47 +1,58 @@
...
+        end_date = datetime.now()
+        batch_size = timedelta(days=5)
```
Comment (sserurich) [inline src/predict/jobs/forecast/models.py:19]: also we can make this batch size configurable(read it from .env)

**Comment by NoahNsimbe** [comment]
Hi @Mnoble-19 
Change the status code from `201` to `200`

File: src/predict/api/controllers/prediction.py [hunk 5]
```diff
@@ -64,30 +46,13 @@ def get_next_24hr_forecasts(device_channel_id, forecast_start_time):
...
+        result = get_forecasts(site_id, db_name='daily_forecasts')
```
Comment (sserurich) [inline src/predict/api/controllers/prediction.py:55]: don't think 'daily_forecasts' is a db_name, might need to reconsider changing the variable name.

**Review by sserurich — COMMENTED** [review]
hi @Mnoble-19 , so far this works , I have tested it and gotten some results, however there some things we need to sort. Some times when running the forecast job, I was getting 
image that message, we can't ignore it. Then most  importantly we need to sychronise and have the training job utelise the same kind of updated setup of site_ids for training the model. perhaps it might solve even the issue I was getting.

**Review by Baalmart — APPROVED** [review]
LGTM, thanks @Mnoble-19

**Pull request merged**